src/game/data.py

```diff
@@ -74,28 +75,21 @@ def load_state(self, initial_state):
 
     def place_piece(self, position, piece_id):
         x, y = get_coordinates(position)
+
         if self.remaining_pieces.count(piece_id) == 1:
             self.grid[y][x] = piece_id
             self.remaining_pieces.remove(piece_id)
         else:
             raise ValueError('Unvalid piece id')
 
     def init_grid(self):
-        grid = []
-        i = 0
-        while i < GameState.grid_size:
-            grid.append([])
-            j = 0
-            while j < GameState.grid_size:
-                grid[i].append('.')
-                j += 1
-            i += 1
-        return grid
+        return [['.' for i in range(GRID_SIZE)] for j in range(GRID_SIZE)]
 
     def init_remaining_pieces(self):
+        return [i+1 for i in range(PIECES_NUMBER)]
```

Comment:
Cool ! Needs spaces around the `+` and the code after the return should be deleted since it's never run.